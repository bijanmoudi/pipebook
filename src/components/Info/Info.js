import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './index.scss';

const Info = ({ items, className, ...restProps }) => {
	return items && Object.keys(items).length ? (
		<ul
			className={classnames('info', { [className]: className })}
			{...restProps}
		>
			{Object.keys(items).map(
				(item, key) =>
					items[item] && (
						<li key={`__info${key}`} className="info__item">
							<span className="info__title">
								<strong>{item}</strong>
								<span className="visuallyhidden">:</span>
							</span>
							<span className="info__content">{items[item]}</span>
						</li>
					)
			)}
		</ul>
	) : null;
};

Info.propTypes = {
	items: PropTypes.object
};

Info.defaultProps = {
	items: null
};

export default Info;
