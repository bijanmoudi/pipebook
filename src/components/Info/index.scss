/*===================================
=            Info Styles            =
===================================*/

@import '../../assets/styles/constants/layout';
@import '../../assets/styles/helpers/colors';
@import '../../assets/styles/helpers/units';

.info {
	font-size: em(14);
	line-height: 1;
	&__item {
		display: flex;
		& + & {
			margin-top: em(23, 14);
		}
	}
	&__title {
		flex-shrink: 0;
		width: em(130, 14);
		padding-right: em(28, 14);
		color: get-color('gray-dark');
		text-align: right;
		font-weight: $semi-bold-font-weight;
		text-transform: capitalize;
		& > * {
			display: inline-block;
			vertical-align: middle;
			font-weight: inherit;
		}
	}
	&__content {
		color: get-color('gray');
	}
}
