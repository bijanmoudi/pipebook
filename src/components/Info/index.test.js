import React from 'react';
import renderer from 'react-test-renderer';
import Info from './Info';

const props = {
	items: {
		hasNull: {
			organization: null,
			groups: 'Mike’s contacts'
		},
		noNull: {
			organization: 'Rockford Linear Actuation',
			groups: 'Mike’s contacts'
		}
	}
};

describe('Testing "Info" component', () => {
	it('Rendering without items', () => {
		const tree = renderer.create(<Info />).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Rendering with items contaning null values', () => {
		const tree = renderer
			.create(<Info items={props.items.hasNull} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('Rendering with items contaning no null value', () => {
		const tree = renderer
			.create(<Info items={props.items.noNull} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
