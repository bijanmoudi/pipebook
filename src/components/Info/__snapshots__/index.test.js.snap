// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing "Info" component Rendering with items contaning no null value 1`] = `
<ul
  className="info"
>
  <li
    className="info__item"
  >
    <span
      className="info__title"
    >
      <strong>
        organization
      </strong>
      <span
        className="visuallyhidden"
      >
        :
      </span>
    </span>
    <span
      className="info__content"
    >
      Rockford Linear Actuation
    </span>
  </li>
  <li
    className="info__item"
  >
    <span
      className="info__title"
    >
      <strong>
        groups
      </strong>
      <span
        className="visuallyhidden"
      >
        :
      </span>
    </span>
    <span
      className="info__content"
    >
      Mike’s contacts
    </span>
  </li>
</ul>
`;

exports[`Testing "Info" component Rendering with items contaning null values 1`] = `
<ul
  className="info"
>
  <li
    className="info__item"
  >
    <span
      className="info__title"
    >
      <strong>
        groups
      </strong>
      <span
        className="visuallyhidden"
      >
        :
      </span>
    </span>
    <span
      className="info__content"
    >
      Mike’s contacts
    </span>
  </li>
</ul>
`;

exports[`Testing "Info" component Rendering without items 1`] = `null`;
